### DISCUSSION TITLE
Error creating a pull request

### DISCUSSION THREAD

**Discussion by isadon** [magit/forge#478] [Support] [answered]
magit-version: `Magit 20220315.309 [>= 3.3.0-git], Git 2.35.1, Emacs 27.2, darwin`
forge: `20220310.1917` installed from melpa

The global git config contains the relevant:
```
[branch]
autoSetupMerge = always
```

So my workflow is as follows: 

1.  Create a branch `test` from `develop` via `b c`
2.  Work on `test` until ready. `branch.test.merge` is `refs/heads/test` `branch.test.remote` is `.`
3. Do `n c p` to create a pull request. `source branch: test` `target branch: develop`

But at step 3 I am getting the error:   `Cannot determine forge repository. No url configured for "."`

I think the "." is referring to the variable `branch.test.remote` being set to `.`.

What I'm expecting is for forge to realize that this `test` branch is a new and local only branch that has NOT been pushed to `origin` and therefore it should:
1. Create `test` on origin and push it.
2. Create the pull request.

The behavior above is what I've normally seen it do some time back but it's no longer doing this. 

Pulling topics and notifications from the repo works fine.

If I first push `test` to origin, creating the pull request works fine from forge.

**Comment by tarsius** [comment, marked as answer]
> The behavior above is what I've normally seen it do some time back but it's no longer doing this.

That seems unlikely because I think that would require a change to `forge-create-pullreq--read-args` and there hasn't been one in over two years.

> Do `n c p` to create a pull request. `source branch: test` `target branch: develop`

I am surprised you are able to do that. `forge-create-pullreq--read-args` is supposed to only offer remote branches for completion (do you observer something else?) and to refuse to let you select something that isn't among the candidates. What completion framework are you using?

**Reply by isadon** [reply]
> I am surprised you are able to do that. forge-create-pullreq--read-args is supposed to only offer remote branches for completion (do you observer something else?) and to refuse to let you select something that isn't among the candidates. What completion framework are you using?

forge-create-pull-req-args does seem to be only offering the remote branches but it doesn't stop me from manually writing out the local branch `test` and using it. The completion framework I am using is `helm` through [prelude](https://prelude.emacsredux.com/en/latest/)

**Reply by tarsius** [reply]
Have you either enabled `helm` globally or set `magit-completing-read-function` (which see)?

**Reply by isadon** [reply]
How would I check either of those two things? In my status bar I see the following so yes I believe Helm is enabled globally.

<img width="827" alt="Emacs Status Bar" src="https://user-images.githubusercontent.com/1881038/160681952-92809286-fbe5-4b1d-8e73-04a5b529f2ab.png">
